tool/lldb_tool.py

```diff
@@ -12,8 +12,109 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """A tool for LLM agents to interact within a LLDB."""
+import logging
+import os
+import subprocess as sp
+
+from experiment import oss_fuzz_checkout
+from experiment.benchmark import Benchmark
+from results import RunResult
 from tool.base_tool import BaseTool
 
+logger = logging.getLogger(__name__)
+
+# The directory in the oss-fuzz image
+JCC_DIR = '/usr/local/bin'
+
 
 class LLDBTool(BaseTool):
-  pass
+  """A tool for LLM agents to interact within a LLDB."""
+
+  def __init__(self, benchmark: Benchmark, name: str, project: str,
+               result: RunResult) -> None:
+    super().__init__(benchmark, name)
+    self.project = project
+    self.result = result
+    self.image_name = self._prepare_project_image()
+    self.container_id = self._start_docker_container()
+
+  def _prepare_project_image(self) -> str:
+    """Prepares the project's OSS-Fuzz docker image and returns the image name.
+    """
+    image_name = f'gcr.io/oss-fuzz/{self.project}'
+    # TODO(maoyi): implement image cache
+    if oss_fuzz_checkout.image_exists_locally(image_name, self.project):
+      logger.info('Using existing project image for %s', self.project)
+      return image_name
+    logger.info('Unable to find existing project image for %s', self.project)
+    command = [
+        'docker', 'build', '-t', image_name,
+        os.path.join(oss_fuzz_checkout.OSS_FUZZ_DIR, 'projects', self.project)
+    ]
+    try:
+      sp.run(command, cwd=oss_fuzz_checkout.OSS_FUZZ_DIR, check=True)
+      logger.info('Successfully build project image for %s', self.project)
+      return image_name
+    except sp.CalledProcessError:
+      logger.info('Failed to build image for %s', self.project)
+      return ''
+
+  def _execute_command(self,
+                       command: list[str],
+                       in_container: bool = False) -> sp.CompletedProcess:
+    """Executes the |command| in subprocess and log output."""
+    result = sp.run(command,
+                    stdout=sp.PIPE,
+                    stderr=sp.PIPE,
+                    check=False,
+                    text=True)
+
+    if in_container:
+      logger.debug(
+          'Executing command (%s) in container %s: Return code %d. STDOUT: %s, '
+          'STDERR: %s', command, self.container_id, result.returncode,
+          result.stdout, result.stderr)
+    else:
+      logger.debug(
+          'Executing command (%s): Return code %d. STDOUT: %s, '
+          'STDERR: %s', command, result.returncode, result.stdout,
+          result.stderr)
+    return result
+
+  def _start_docker_container(self) -> str:
```

Comment:
Could we reuse
https://github.com/google/oss-fuzz-gen/blob/f08dd922b7fd2a2a1491fda36731e43c97d919ea/tool/container_tool.py#L108